### PR TITLE
Support complex numbers

### LINKED ISSUE #96: Support complex-valued matrices?
We use an LDL' factorization in Convex.jl to [check for PSD-ness](https://github.com/jump-dev/Convex.jl/pull/457) of real- and complex-valued hermitian matrices.

Being able to use this package with complex values and arbitrary arithmetic would greatly simplify things for us.

### PR DESCRIPTION
**Pull request by amontoison**
close #96
@mtanneau, @dpo

### CODE REVIEW

**Comment by amontoison** [comment]
We need to merge https://github.com/JuliaSmoothOptimizers/AMD.jl/pull/51 before and do a new release.
`amd(Hermitian(A))` is not defined otherwise.

File: src/LDLFactorizations.jl [hunk 1]
```diff
@@ -438,7 +438,7 @@ The components of the factorization can be accessed via `getproperty`:
 In order to avoid zero pivots during the factorization, the user can regularize the matrix by modifying 
 `LDL.r1` for the `LDL.n_d` first pivots and `LDL.r2` for the other pivots with tolerance `LDL.tol`.
 """
-mutable struct LDLFactorization{T <: Real, Ti <: Integer, Tn <: Integer, Tp <: Integer}
+mutable struct LDLFactorization{T <: Number, Ti <: Integer, Tn <: Integer, Tp <: Integer}
```
Comment (dpo) [inline src/LDLFactorizations.jl:441]: Hoping to get rid of `T <: Number` and just leave it as `T`.

Comment (dpo) [reply src/LDLFactorizations.jl:441]: Yes. Actually, I realized this morning that `AbstractInterval <: Real`, so it should work with this PR out of the box 🤞 .

**Comment by dpo** [comment]
@JSOBot runbenchmarks

**Comment by dpo** [comment]
@JSOBot runbenchmarks

**Comment by amontoison** [comment]
@JSOBot runbenchmarks

**Comment by MonssafToukal** [comment]
@JSOBot runbenchmarks

**Comment by amontoison** [comment]
I opened a PR to fix the compat issue: https://github.com/osqp/QDLDL.jl/pull/8

**Comment by amontoison** [comment]
@JSOBot runbenchmarks

**Comment by amontoison** [comment]
@MonssafToukal, the compat issue is fixed, we can run again the benchmarks.

**Comment by amontoison** [comment]
@JSOBot runbenchmarks

**Comment by amontoison** [comment]
@JSOBot runbenchmarks

**Comment by amontoison** [comment]
@MonssafToukal, is it possible to always update the general registry when the JSOBot is called?

**Comment by MonssafToukal** [comment]
@amontoison Interesting. I can add an `update` to the script. Thanks for letting me know.

**Comment by amontoison** [comment]
@JSOBot runbenchmarks

**Comment by MonssafToukal** [comment]
@JSOBot runbenchmarks

**Comment by amontoison** [comment]
> Benchmark results: https://gist.github.com/cc383a1a154f202ecab0f37e7750d2e5

We don't have performance regression.
LGTM

**Comment by dpo** [comment]
thank you!

**Pull request merged**